### PR TITLE
feat: adding show() method to python execution results

### LINKED ISSUE #1216: Add `show` method to python execution results
## Summary

Be able to print out results without needing to call `to_polars` or `to_pandas` first.

### PR DESCRIPTION
**Pull request by adhish20**
Fixes #1216

### CODE REVIEW

**Comment by adhish20** [comment]
Output from the example :

pyarrow.Table
A: int64
fruits: large_string
B: int64
cars: large_string
----
A: [[1,2,5]]
fruits: [["banana","banana","banana"]]
...
pyarrow.Table
A: int64
fruits: string
B: int64
cars: string
-- schema metadata --
pandas: '{"index_columns": [{"kind": "range", "name": null, "start": 0, "' + 677
----
A: [[1,2,5]]
fruits: [["banana","banana","banana"]]
B: [[5,4,1]]
cars: [["beetle","audi","beetle"]]

File: py-glaredb/src/session.rs [hunk 1]
```diff
@@ -171,4 +175,24 @@ impl PyExecutionResult {
             Ok(result)
         })
     }
+
+    #[allow(clippy::wrong_self_convention)] // this is consistent with other python API's
```
Comment (universalmind303) [inline py-glaredb/src/session.rs:179]: i don't think we need this for `show`

File: py-glaredb/src/logical_plan.rs [hunk 2]
```diff
@@ -60,6 +60,9 @@ impl PyLogicalPlan {
     fn to_pandas(&self, py: Python) -> PyResult<PyObject> {
         self.collect(py)?.to_pandas(py)
     }
+    fn show(&self, py: Python, show_metadata: bool, n_cols: u64) -> PyResult<()> {
+        self.collect(py)?.show(py, show_metadata, n_cols)
+    }
```
Comment (universalmind303) [inline py-glaredb/src/logical_plan.rs:65]: shouldn't this be `n_rows` not `n_cols`?

Comment (adhish20) [reply py-glaredb/src/logical_plan.rs:65]: the to_string API from pyarrow supports only number of columns and not number of rows

https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table.to_string

File: py-glaredb/examples/show.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        "A": [1, 2, 3, 4, 5],
+        "fruits": ["banana", "banana", "apple", "apple", "banana"],
+        "B": [5, 4, 3, 2, 1],
+        "cars": ["beetle", "audi", "beetle", "beetle", "beetle"],
+    }
+)
+
+con = glaredb.connect()
+
+con.sql("select * from df where fruits = 'banana'").show(True, 4);
```
Comment (universalmind303) [inline py-glaredb/examples/show.py:27]: I think the signature should use kwargs. 

`show(n_rows, show_metadata=False)`

such as 
```python
x.show(10)
x.show(n_rows=10)
x.show(10, show_metadata=True)
x.show(10, show_metadata=False)
x.show(n_rows=10, show_metadata=False)
```

File: py-glaredb/src/logical_plan.rs [hunk 4]
```diff
@@ -60,6 +60,9 @@ impl PyLogicalPlan {
     fn to_pandas(&self, py: Python) -> PyResult<PyObject> {
         self.collect(py)?.to_pandas(py)
     }
+    fn show(&self, py: Python, show_metadata: bool, n_cols: u64) -> PyResult<()> {
+        self.collect(py)?.show(py, show_metadata, n_cols)
```
Comment (universalmind303) [inline py-glaredb/src/logical_plan.rs:64]: for the logical_plan, we should push down the `n_rows` to the plan. No sense in over fetching data 😃

**Review by universalmind303 — CHANGES_REQUESTED** [review]

File: py-glaredb/src/session.rs [hunk 5]
```diff
@@ -171,4 +175,24 @@ impl PyExecutionResult {
...
+
+            let key_vals: &[(&str, PyObject)] = &[
+                ("show_metadata", show_metadata.to_object(py)),
+                ("preview_cols", n_cols.to_object(py)),
+            ];
+            let dict = key_vals.into_py_dict(py);
+            let result = table.call_method(py, "to_string", (), Some(dict))?;
+            println!("{}", result);
+            Ok(())
+        })
```
Comment (universalmind303) [inline py-glaredb/src/session.rs:195]: i think instead of going in to pyarrow, we can just print the execution result as is.

Comment (adhish20) [reply py-glaredb/src/session.rs:195]: Makes sense. I will look into that

File: py-glaredb/examples/show.py (hunk 3 above)
Comment (universalmind303) [reply py-glaredb/examples/show.py:27]: On second thought... I think this shouldn't have any args. I think having similar functionality to duckdb's [show](https://duckdb.org/docs/api/python/relational_api) would be a good starting point. 

if you want to limit the n_rows, you can just add `limit x` to your sql query.

File: py-glaredb/src/logical_plan.rs (hunk 2 above)
Comment (universalmind303) [reply py-glaredb/src/logical_plan.rs:65]: On second thought... I think this shouldn't have any args. I think having similar functionality to duckdb's [show](https://duckdb.org/docs/api/python/relational_api) would be a good starting point.

if you want to limit the n_rows, you can just add limit x to your sql query.

File: py-glaredb/examples/show.py (hunk 3 above)
Comment (adhish20) [reply py-glaredb/examples/show.py:27]: cool. will remove the args

**Comment by adhish20** [comment]
Updated Output looks like

+---+--------+---+--------+
| A | fruits | B | cars   |
+---+--------+---+--------+
| 1 | banana | 5 | beetle |
| 2 | banana | 4 | audi   |
| 5 | banana | 1 | beetle |
+---+--------+---+--------+

+---+--------+---+--------+
| A | fruits | B | cars   |
+---+--------+---+--------+
| 1 | banana | 5 | beetle |
| 2 | banana | 4 | audi   |
| 5 | banana | 1 | beetle |
+---+--------+---+--------+

**Pull request merged**